packages/@angular/cli/models/config.ts

```diff
@@ -5,7 +5,8 @@ import * as chalk from 'chalk';
 import * as fs from 'fs';
 import * as path from 'path';
 
-export const CLI_CONFIG_FILE_NAME = 'angular-cli.json';
+export const CLI_CONFIG_FILE_NAME = '.angular-cli.json';
+const CLI_CONFIG_FILE_NAME_ALT = 'angular-cli.json';
```

Comment:
I spoke with @hansl about this, and we think we should silently fallback to the old file name. This means fewer changes are needed, but newly generated projects will generate with the new file name.